Kalenteri/Frontend/Pages/Home.razor

```diff
@@ -1,39 +1,198 @@
-﻿@page "/"
+﻿@using Kalenteri.Backend.Models
+
+@page "/"
+@rendermode InteractiveServer
 
 <PageTitle>Kalenteri</PageTitle>
 
-<h1>Livonsaaren Osuuspuutarha</h1>
+<p>Käyttäjä: <b>@userId</b></p>
+<p>Jakopaikka: <b>@location</b></p>
+
+<h2 class="month-title">@MonthToString()</h2>
 
-@foreach (var date in dates)
-{
-    <button class="btn btn-primary" style="@StyleForState(date.State)"
-    @onclick="() => HandleClick(date.Day)">@date.State</button>
-}
+<div class="calendar">
+    <span>Ma</span>
+    <span>Ti</span>
+    <span>Ke</span>
+    <span>To</span>
+    <span>Pe</span>
+    <span>La</span>
+    <span>Su</span>
+    @for (int pad = 0; pad < GetFirstWeekPadding(); pad++)
+    {
+        <span class="week-day-padding" />
+    }
+    @for (int day = 1; day < GetMonthDayCount(); day++)
+    {
+        var order = orders.Single(x => x.Identifier == "Testi");
+        var delivery = order.Boxes.SingleOrDefault(x => x.Delivery.Day == day && x.Delivery.Month == month);
+
+        if (delivery != null)
+        {
+            <button class="btn" @onclick="@(_ => HandlePickUpClick(delivery.Delivery))"
+                style="@StyleForState(delivery.PickUp == null)">@day</button>
+        }
+        else
+        {
+            <button class="btn empty-btn" disabled>@day</button>
+        }
+    }
+</div>
+
+<div class="month-navigation">
+    <button class="btn month-nav-btn" @onclick="DecreaseMonth">Edellinen</button>
+    <button class="btn month-nav-btn" @onclick="IncreaseMonth">Seuraava</button>
+</div>
 
 @code {
-    private int currentCount = 0;
-    private List<Date> dates = new List<Date>([
-    new Date { Day = 1, State = true },
-new Date { Day = 2, State = true },
-new Date { Day = 3, State = false },
-new Date { Day = 4, State = true },
-new Date { Day = 5, State = false }
-    ]);
+    private int month = DateTime.Now.Month;
+    private string userId = "Testi";
+    private string location = "Testikatu 16, 12345 Testilandia";
 
-    private string StyleForState(bool state)
+    private void DecreaseMonth()
+    {
+        if (month != 1)
+        {
+            month--;
+        }
+    }
+
+    private void IncreaseMonth()
+    {
+        if (month != 12)
+        {
+            month++;
+        }
+    }
+
+    private string MonthToString()
     {
-        if (state) return "background:lightgreen";
-        return "background:red";
+        switch (month)
```

Comment:
Tällein sais noi nimet kaivettua ilman switchailua. 

    string monthName = CultureInfo.CreateSpecificCulture("fi").DateTimeFormat.GetAbbreviatedMonthName(8);

ja jotenkin näin isolla alkukirjaimella 

    string monthName = CultureInfo.CreateSpecificCulture("fi").TextInfo.ToTitleCase(CultureInfo.CreateSpecificCulture("fi").DateTimeFormat.GetAbbreviatedMonthName(8));

Reply:
https://github.com/codevictory/Kalenteri/pull/1/commits/0058f571cc6e7eec61fe5725fff468f8ee37396a